docs/source/guides/quickstart.md

```diff
@@ -1,48 +1,195 @@
+can be passed into any `Character`'s init. The `known_nodes` will inform your character of all of the nodes
+they know about network-wide, then kick-off the automated node-discovery loop:
+
+```python
+alice = Alice(known_nodes=[ursula, another_ursula], ...)
+```
+
+## Alice: Grant Access to a Secret
+
+```python
+from nucypher.characters.lawful import Alice, Bob, Ursula
+from nucypher.network.middleware import RestMiddleware
+from nucypher.blockchain.eth.registry import InMemoryContractRegistry
+
+# Application Side-Channel
+# --------------------------
+# bob_encrypting_key = <Side Channel>
+# bob_verifying_key = <Side Channel>
+
+ursula = Ursula.from_seed_and_stake_info(seed_uri='https://0.0.0.0:9151', federated_only=False)
+alice = Alice(known_nodes=[ursula],
+              checksum_address="0x287A817426DD1AE78ea23e9918e2273b6733a43D", 
+              registry=InMemoryContractRegistry.from_latest_publication(),
+              network_middleware=RestMiddleware())
+bob = Bob.from_public_keys(verifying_key=bob_verifying_key, encrypting_key=bob_encrypting_key)
+```
+
+```python
+from datetime import datetime
+import maya
+
+#
+# Grant
+#
+
+policy_end_datetime = maya.now() + datetime.timedelta(days=5)  # Five days from now
+policy = alice.grant(bob,
+                     label=b'my-secret-stuff',
+                     m=2, n=3,
+                     expiration=policy_end_datetime)
+
+policy_encrypting_key = policy.public_key
 ```
 
-Replace `<GETH_DEV_ADDRESS>` with the geth node's public checksum address.
+## Enrico: Encrypt a Secret
+
+```python
+from nucypher.characters.lawful import Enrico
+
+# Application Side-Channel
+# --------------------------
+# policy_encrypting_key = <Side Channel>
+
+enrico = Enrico(policy_encrypting_key=policy_encrypting_key)
+ciphertext, signature = enrico.encrypt_message(message=b'Peace at dawn.')
+```
+    
+Note that Alice can get the public key even before creating the policy.
+From this moment on, any Data Source (Enrico) that knows the public key
+can encrypt data originally intended for Alice, but can be shared with
+any Bob that Alice grants access.
+
+`policy_pubkey = alice.get_policy_encrypting_key_from_label(label)`
+
+
+## Bob: Decrypt a Secret
+
+```python
+from nucypher.characters.lawful import Alice, Bob, Enrico, Ursula
+from nucypher.blockchain.eth.registry import InMemoryContractRegistry
+from nucypher.network.middleware import RestMiddleware
+
+# Application Side-Channel
+# --------------------------
+# label = <Side Channel>
+# ciphertext = <Side Channel>
+# policy_encrypting_key = <Side Channel>
+# alice_verifying_key = <Side Channel>
+
+# Everyone!
+ursula = Ursula.from_seed_and_stake_info(seed_uri='https://0.0.0.0:9151', federated_only=False)
+alice = Alice.from_public_keys(verifying_key=alice_verifying_key)
+bob = Bob(known_nodes=[ursula],
+          checksum_address="0xc080708026a3a280894365efd51bb64521c45147",
```

Comment:
The address in this example is not a checksum one

Reply:
:heavy_check_mark: